src/test/java/org/folio/qm/service/impl/MarcRecordsServiceImplTest.java

```diff
@@ -0,0 +1,112 @@
+package org.folio.qm.service.impl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.when;
+
+import static org.folio.qm.utils.TestUtils.EXISTED_INSTANCE_ID;
+import static org.folio.qm.utils.TestUtils.QM_EMPTY_FIELDS;
+import static org.folio.qm.utils.TestUtils.VALID_JOB_EXECUTION_ID;
+import static org.folio.qm.utils.TestUtils.VALID_PARSED_RECORD_DTO_ID;
+import static org.folio.qm.utils.TestUtils.readQuickMarc;
+
+import java.util.Collections;
+import java.util.UUID;
+import java.util.function.Predicate;
+
+import io.github.glytching.junit.extension.random.Random;
+import io.github.glytching.junit.extension.random.RandomBeansExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mapstruct.ap.internal.util.Strings;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import org.folio.qm.client.SRMChangeManagerClient;
+import org.folio.qm.converter.QuickMarcToParsedRecordDtoConverter;
+import org.folio.qm.domain.dto.CreationStatus;
+import org.folio.qm.domain.dto.QuickMarcFields;
+import org.folio.qm.domain.entity.RecordCreationStatus;
+import org.folio.qm.domain.entity.RecordCreationStatusUpdate;
+import org.folio.qm.mapper.CreationStatusMapper;
+import org.folio.qm.service.CreationStatusService;
+import org.folio.qm.service.ValidationService;
+import org.folio.qm.util.ChangeManagerPayloadUtils;
+import org.folio.qm.util.JobExecutionProfileProperties;
+import org.folio.rest.jaxrs.model.InitJobExecutionsRqDto;
+import org.folio.rest.jaxrs.model.InitJobExecutionsRsDto;
+import org.folio.rest.jaxrs.model.JobExecution;
+import org.folio.rest.jaxrs.model.JobProfileInfo;
+import org.folio.spring.FolioExecutionContext;
+
+@ExtendWith({
+  MockitoExtension.class,
+  RandomBeansExtension.class
+})
+class MarcRecordsServiceImplTest {
+
+  @InjectMocks
+  private MarcRecordsServiceImpl service;
+  @Mock
+  private ValidationService validationService;
+  @Mock
+  private FolioExecutionContext context;
+  @Mock
+  private SRMChangeManagerClient srmClient;
+  @Mock
+  private CreationStatusService statusService;
+  @Mock
+  private CreationStatusMapper statusMapper;
+  @Mock
+  private JobExecutionProfileProperties props;
+  @Spy
+  private QuickMarcToParsedRecordDtoConverter converter;
+
+
+  @Test
+  void shouldNotThrowError(@Random JobProfileInfo profile) {
```

Comment:
please use the name which describe verification goals more specifically

Reply:
renamed to shouldRemoveFieldsWhenEmptyContent. Thanks for note.